update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+// update in standalone clusters), during the update.
+type ClusterVersionStatusInsight struct {
+	// resource is the ClusterVersion resource that represents the control plane
+	// +required
+	Resource ResourceRef `json:"resource"`
+
+	// assessment is the assessment of the control plane update process
+	// +required
+	Assessment ControlPlaneAssessment `json:"assessment"`
+
+	// versions contains the original and target versions of the upgrade
+	// +required
+	Versions ControlPlaneUpdateVersions `json:"versions"`
+
+	// completion is a percentage of the update completion (0-100)
+	// +required
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	Completion uint8 `json:"completion"`
+
+	// startedAt is the time when the update started
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	StartedAt metav1.Time `json:"startedAt"`
+
+	// completedAt is the time when the update completed
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	CompletedAt metav1.Time `json:"completedAt,omitempty"`
+
+	// estimatedCompletedAt is the estimated time when the update will complete
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	EstimatedCompletedAt metav1.Time `json:"estimatedCompletedAt,omitempty"`
+
+	// Conditions provides details about the control plane update
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ClusterOperatorStatusInsightConditionType are types of conditions that can be reported on ClusterOperator status insights
+type ClusterOperatorStatusInsightConditionType string
+
+const (
+	// Updating condition communicates whether the ClusterOperator is updating
+	ClusterOperatorStatusInsightUpdating ClusterOperatorStatusInsightConditionType = "Updating"
+	// Healthy condition communicates whether the ClusterOperator is considered healthy
+	ClusterOperatorStatusInsightHealthy ClusterOperatorStatusInsightConditionType = "Healthy"
+)
+
+// ClusterOperatorUpdatingReason are well-known reasons for the Updating condition on ClusterOperator status insights
+type ClusterOperatorUpdatingReason string
+
+const (
+	// Updated is used with Updating=False when the ClusterOperator finished updating
+	ClusterOperatorUpdatingReasonUpdated ClusterOperatorUpdatingReason = "Updated"
+	// Pending is used with Updating=False when the ClusterOperator is not updating and is still running previous version
+	ClusterOperatorUpdatingReasonPending ClusterOperatorUpdatingReason = "Pending"
+	// Progressing is used with Updating=True when the ClusterOperator is updating
+	ClusterOperatorUpdatingReasonProgressing ClusterOperatorUpdatingReason = "Progressing"
+	// CannotDetermine is used with Updating=Unknown
+	ClusterOperatorUpdatingCannotDetermine ClusterOperatorUpdatingReason = "CannotDetermine"
```

Comment:
nit: If we already have `Unknown`, why pivot to `CannotDetermine` here?  Can't we recycle `Unknown` to avoid the need to translate between similar terms?

Reply:
`Unknown` is condition.Status, `CannotDetermine` is condition.Reason. I agree that it feels redundant but kube conditions [want `reason` to be used](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties):

> Use of the Reason field is required.

I originally had reasons like `MissingProgressing` and `UnknownProgressing` (and other reasons why we cannot figure out stuff when we rely on conditions) but felt like it's not necessary in the first iteration and more granular reasons for unknown can be added later. 

Or actually, am I reading this  comment wrong and you simply propose `status=Unknown` and `reason=Unknown`?

Reply:
I guess I'm saying `CannotDetermine` doesn't add much context beyond what `Unknown` already brings.  But as of 995b1612ca, you have Godocs with:

> This covers many different actual reasons such as missing or Unknown Progressing condition on ClusterVersion, but it does not seem useful to track the individual reasons to that granularity for Updating=Unknown

and that sounds reasonable to me.  I'll mark this thread resolved.